setup.py

```diff
@@ -869,6 +875,148 @@ def prefix(directory, files):
     extra_objects=zstd_extra_objects,
 )
 
+# SZ3 library and its hdf5 filter
+# Generate needed version include file.
+with open(os.path.join("src", "SZ3", "include", "SZ3", "version.hpp"), 'w') as f:
+    f.write("//\n")
+    f.write("// DO NOT MODIFY version.hpp, it is automatically generated by cmake\n")
+    f.write("// The version is controlled in CMakeLists.txt\n")
+    f.write("//\n")
+    f.write("\n")
+    f.write("#ifndef SZ3_VERSION_HPP\n")
+    f.write("#define SZ3_VERSION_HPP\n")
+    f.write("\n")
+    f.write('#define SZ3_NAME "SZ3"\n')
+    f.write('#define SZ3_VER  "3.1.7"\n')
+    f.write("#define SZ3_VER_MAJOR 3\n")
+    f.write("#define SZ3_VER_MINOR 1\n")
+    f.write("#define SZ3_VER_PATCH 7\n")
+    f.write("#define SZ3_VER_TWEAK \n")
+    f.write("\n")
+    f.write("#endif //SZ3_VERSION_HP\n")
```

Comment:
Can't this file be put in `SZ3_extra/`?

Reply:
I had already tried, however the code explicitly import from "SZ3/version.hpp".

I can try to add a repository SZ3 containing version.hpp inside SZ3_extra to see if it works.